src/transformers/models/t5/modeling_tf_t5.py

```diff
@@ -1481,6 +1481,10 @@ def prepare_inputs_for_generation(
             encoder_outputs, past_key_values = past, None
         else:
             encoder_outputs, past_key_values = past[0], past[1]
+        if "encoder_hidden_states" in kwargs:
```

Comment:
Do we need to do this just for T5 or for other models as well?

Reply:
This stuff is T5-specific only. In that case, during decoding when:
- `encoder_outputs` are fed into the model solely without `encoder_attentions` and `encoder_hidden_states`, and
- `output_attentions=True` or `output_hidden_states=True`,

the error occurs in this part of code.

```
elif isinstance(inputs["encoder_outputs"], tuple):
    last_hidden_state = inputs["encoder_outputs"][0]
    hidden_states = None
    attentions = None
    idx = 0
    if inputs["output_hidden_states"]:
        idx += 1
        hidden_states = inputs["encoder_outputs"][idx]
    if inputs["output_attentions"]:
        idx += 1
        attentions = inputs["encoder_outputs"][idx]

    inputs["encoder_outputs"] = TFBaseModelOutput(
        last_hidden_state=last_hidden_state,
        hidden_states=hidden_states,
        attentions=attentions,
    )
```
This results from that neither `output_attentions` nor `output_hidden_states` is present in the inputs["output_hidden_states"] because encoder pass is done if and only if -- `if inputs["encoder_outputs"] is None`.